### ISSUE TITLE
Failed trackable goes to offline when subscriber is stopped

### ISSUE THREAD

**Issue by davyskiba** [ably/ably-asset-tracking-android#802]
Stopping the subscriber uses [StopConnectionWorker](https://github.com/ably/ably-asset-tracking-android/blob/2897a8c30faa86e4452acfed763f3831990a5163/subscribing-sdk/src/main/java/com/ably/tracking/subscriber/workerqueue/workers/StopConnectionWorker.kt#L27) which calls [notifyAssetIsOffline](https://github.com/ably/ably-asset-tracking-android/blob/2897a8c30faa86e4452acfed763f3831990a5163/subscribing-sdk/src/main/java/com/ably/tracking/subscriber/CoreSubscriber.kt#L163) function which changes trackable state to `Offline.` 
According to [this](https://github.com/ably/ably-asset-tracking-android/blob/2897a8c30faa86e4452acfed763f3831990a5163/core-sdk/src/main/java/com/ably/tracking/ConfigurationModels.kt#L28) comment, `Failed` state is supposed to be final and this transition should not occur.

**Comment by QuintinWillison** [comment]
I agree that there's something not right with the `notifyAssetIsOffline()` method of `DefaultCoreSubscriber` because it seems to break the constraints of property/state update+emit flows established elsewhere. I've added a `TODO` comment in https://github.com/ably/ably-asset-tracking-android/pull/855.
